### DISCUSSION TITLE
Can CSLA.NET use Java Hibernate datasource?

### DISCUSSION THREAD

**Discussion by aprokopyev** [MarimerLLC/csla#2414] [Questions]
Hello Rockford,

I like nHibernate ORM very much, but unfortunately it is not as feature rich as original Hibernate on JVM.
While I prefer to do all GUI works on DotNet using DevExpress components and your CSLA.net  I am thinking about making data access layer on JVM using true Java Hibernate OGM extensions may be using modern languages for JVM like Kotlin or Scala.

Please let me know, what is the best way to connect CSLA.NET data portal to Java Hibernate classes for DAL?

May be using 

1) Remoting like gRPC ?

2) Or some framework bridge like https://libraries.io/nuget/JCOBridge ?

3) Or may be even RemObjects Hydra or Mercury:
https://hydra.remobjects.com/hydra/default.aspx
https://www.elementscompiler.com/elements/mercury/
It seems they can mix DotNet and Java in the same project.

In result I need to get something like this:
DevExpress GUI -> CSLA.NET on .NET v5+ -> JVM DAL using Hibernate OGM -> modern noSQL DBMS

**Comment by rockfordlhotka** [comment]
I think the challenge here is that the Java ORM will need to run in a separate process from the .NET code. So I would _guess_ that you need to create a service API around your Java ORM, exposing services than can be invoked from your .NET code.

CSLA itself won't really care about this. Your data portal operation methods (create,fetch,update,etc.) will invoke those services instead of talking directly to a database. That is a valid implementation.
